packages/flutter/lib/src/widgets/selection_container.dart

```diff
@@ -0,0 +1,207 @@
+    return widget.delegate.dispatchSelectionEvent(event);
+  }
+
+  @override
+  SelectionGeometry get value => widget.delegate.value;
+
+  @override
+  Matrix4 getTransformTo(RenderObject? ancestor) {
+    return context.findRenderObject()!.getTransformTo(ancestor);
+  }
+
+  @override
+  Size get size => (context.findRenderObject()! as RenderBox).size;
+
+  @override
+  void dispose() {
+    _registeredSelectables.clear();
+    widget.delegate.removeListener(_onSelectionGeometryChange);
+    widget.delegate._selectionContainerContext = null;
+    _notifier.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return SelectionRegistrarScope(
+      registrar: widget.delegate,
+      child: widget.child,
+    );
+  }
+}
+
+/// An inherited widget to host a [SelectionRegistrar] for the subtree.
+///
+/// Use [SelectionRegistrarScope.maybeOf] to get the SelectionRegistrar from
+/// a context.
+///
+/// This widget is automatically created as part of [SelectionContainer] and
+/// is generally not used directly, except for disabling selection for a part
+/// of subtree. In that case, one can wrap the subtree with
+/// [SelectionRegistrarScope.disabled].
+class SelectionRegistrarScope extends InheritedWidget {
+  /// Creates a selection registrar scope that host the [registrar].
+  const SelectionRegistrarScope({
+    super.key,
+    required SelectionRegistrar this.registrar,
+    required super.child,
+  }) : assert(registrar != null);
+
+  /// Creates a selection registrar scope that disables selection for the
+  /// subtree.
+  const SelectionRegistrarScope.disabled({
+    super.key,
+    required super.child,
+  }) : registrar = null;
+
+  /// Gets the immediate ancestor [SelectionRegistrar] of the [BuildContext].
+  ///
+  /// If this returns null, either there is no [SelectionRegistrarScope] above
+  /// the [BuildContext] or the immediate [SelectionRegistrarScope] is not
+  /// enabled.
+  static SelectionRegistrar? maybeOf(BuildContext context) {
+    final SelectionRegistrarScope? scope = context.dependOnInheritedWidgetOfExactType<SelectionRegistrarScope>();
+    return scope?.registrar;
+  }
+
+  /// The [SelectionRegistrar] hosted by this widget.
+  final SelectionRegistrar? registrar;
+
+  @override
+  bool updateShouldNotify(SelectionRegistrarScope oldWidget) {
+    return oldWidget.registrar != registrar;
+  }
+}
+
+/// A delegate to handle selection events for a [SelectionContainer].
+///
+/// This delegate needs to implement [SelectionRegistrar] to register
+/// [Selectable]s in the [SelectionContainer] subtree.
+abstract class SelectionContainerDelegate implements SelectionHandler, SelectionRegistrar {
+  /// The context of the selection container that holds this delegate.
+  ///
+  /// This is useful when retrieving layout geometry of the container.
+  @protected
+  BuildContext get selectionContainerContext => _selectionContainerContext!;
+  BuildContext? _selectionContainerContext;
```

Comment:
The way the BuildContext is injected into this Delegate seems ... iffy. What is that actually needed for?

Reply:
it needs the layout information